src/decode.rs

```diff
@@ -4388,8 +4377,8 @@ pub(crate) unsafe fn rav1d_decode_frame_init(
                 }
                 f.frame_thread.pal_idx_sz = pal_idx_sz;
             }
-        } else if !f.frame_thread.pal.is_null() {
-            rav1d_freep_aligned(&mut f.frame_thread.pal as *mut *mut [[u16; 8]; 3] as *mut c_void);
+        } else if !f.frame_thread.pal.is_empty() {
+            let _ = mem::take(&mut f.frame_thread.pal);
             rav1d_freep_aligned(&mut f.frame_thread.pal_idx as *mut *mut u8 as *mut c_void);
```

Comment:
Is `pal_idx` a pointer into `pal`? If so, we shouldn't be freeing it.

Reply:
Nevermind, I see that it is allocated separately. Weird naming. I guess we need to have another PR to switch it over too.